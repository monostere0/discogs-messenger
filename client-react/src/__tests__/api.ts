import {
  getIdentity,
  getMessages,
  getMessage,
} from '../api';
import fetch from '../fetch';

jest.mock('../fetch', () => ({
  __esModule: true,
  default: jest.fn(() => {
    return Promise.resolve({
      json() {
        return Promise.resolve({ foo: 'bar' });
      }
    })  
  }),
}));

describe('src/api', () => {
  it('getIdentity()', async () => {
    const response = await getIdentity();
    expect(fetch).toHaveBeenCalledWith('../api/identity');
    expect(response).toMatchSnapshot();
  });
  it('getMessages()', async () => {
    const response = await getMessages();
    expect(fetch).toHaveBeenCalledWith('../api/messages');
    expect(response).toMatchSnapshot();
  });
  it('getMessage(orderId)', async () => {
    const response = await getMessage('44-112-2');
    expect(fetch).toHaveBeenCalledWith('../api/messages/44-112-2');
    expect(response).toMatchSnapshot();
  });
});
