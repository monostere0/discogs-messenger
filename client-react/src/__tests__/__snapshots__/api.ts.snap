// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/api getIdentity() 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`src/api getMessage(orderId) 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`src/api getMessages() 1`] = `
Object {
  "foo": "bar",
}
`;
